test/e2e/e2e_test.go

```diff
@@ -49,6 +49,13 @@ var _ = Describe("e2e", func() {
 		_ = device.Unregister()
 		_ = device.Remove()
 	})
+
+	JustAfterEach(func() {
```

Comment:
This change is great, but in this case, AfterFailed is not going to be executed, needs to be added *after* this `JustAfterEach()`

So, I would say that move AfterFailed function to the line 59, and if race happens, it'll be logged correctly :-)